src/transformers/pipelines/__init__.py

```diff
@@ -367,6 +372,9 @@ def pipeline(
 
     task_class, model_class = targeted_task["impl"], targeted_task[framework]
 
+    # Retrieve use_auth_token and add it to model_kwargs to be used in .from_pretrained
+    model_kwargs["use_auth_token"] = model_kwargs.get("use_auth_token", use_auth_token)
+
```

Comment:
model_kwargs can't contain the `use_auth_token` key since you took it out of the model_kwargs in the signature. So this line should just be
```suggestion
    # Retrieve use_auth_token and add it to model_kwargs to be used in .from_pretrained
    model_kwargs["use_auth_token"] = use_auth_token

```

Reply:
are you sure? `model_kwargs` are a `dict` and not `**kwargs`. So the named parameter is not removing it from `**kwargs`, since it is in `model_kwargs={'use_auth_token':'xxx'}`. 

I tested both examples in the comment.

Reply:
Indeed, sorry about that.